unstructured/embed/huggingface.py

```diff
@@ -0,0 +1,69 @@
+from typing import List, Optional
+
+import numpy as np
+from langchain.embeddings import HuggingFaceEmbeddings
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.embed.interfaces import BaseEmbeddingEncoder
+from unstructured.ingest.error import EmbeddingEncoderConnectionError
+from unstructured.utils import requires_dependencies
+
+
+class HuggingFaceEmbeddingEncoder(BaseEmbeddingEncoder):
+    def __init__(
+        self,
+        model_name: Optional[str] = "sentence-transformers/all-MiniLM-L6-v2",
+        model_kwargs: Optional[dict] = {"device": "cpu"},
+        encode_kwargs: Optional[dict] = {"normalize_embeddings": False},
+        cache_folder: Optional[dict] = None,
+    ):
+        self.model_name = model_name
+        self.model_kwargs = model_kwargs
+        self.encode_kwargs = encode_kwargs
+        self.cache_folder = cache_folder
+
+        self.initialize()
+
+    @EmbeddingEncoderConnectionError.wrap
+    @requires_dependencies(
+        ["langchain", "huggingface"],
+        extras="huggingface",
```

Comment:
```suggestion
        extras="embed-huggingface",
```
This string signifies which extra that we defined can be utilized to install each of the potentially missing `pip install`s altogether. 

For our case, for it to not be confused with the existing huggingface extra, let's name it as `embed-huggingface`. 

Since we name it as `embed-huggingface`, now we need to set it up. To do that, we will need to delete these:
- https://github.com/Unstructured-IO/unstructured/blob/88592e5c522bdbd81f2ec8058c28c7d237690bb8/requirements/langchain.in
- https://github.com/Unstructured-IO/unstructured/blob/88592e5c522bdbd81f2ec8058c28c7d237690bb8/requirements/langchain.txt
- https://github.com/Unstructured-IO/unstructured/blob/88592e5c522bdbd81f2ec8058c28c7d237690bb8/setup.py#L161

(and remove any other change we introduced for an extra named "langchain")

And we need to create them again for `embed-huggingface`, with relevant requirements. For example, `embed-huggingface.in` should look like:
```
-c constraints.in
-c base.txt
langchain
sentence_transformers
```

Reply:
So after setting up the extra correctly, the CI job (test unit) should be able to install all requirements via the name of the dependency